File: MiniDOMFinanziaria/Finanziaria/RichiesteDeroghe/CRichiestaDerogaCursor.vb
Imports minidom
Imports minidom.Databases
Imports minidom.Sistema
Imports minidom.Anagrafica



Partial Public Class Finanziaria


    Public Class CRichiestaDerogaCursor
        Inherits DBObjectCursorPO(Of CRichiestaDeroga)

        Private m_IDCliente As New CCursorField(Of Integer)("IDCliente")
        Private m_NomeCliente As New CCursorFieldObj(Of String)("NomeCliente")
        Private m_StatoRichiesta As New CCursorField(Of StatoRichiestaDeroga)("StatoRichiesta")
        Private m_DataRichiesta As New CCursorField(Of Date)("DataRichiesta")
        Private m_IDRichiedente As New CCursorField(Of Integer)("IDRichiedente")
        Private m_NomeRichiedente As New CCursorFieldObj(Of String)("NomeRichiedente")
        Private m_MotivoRichiesta As New CCursorFieldObj(Of String)("MotivoRichiesta")
        Private m_IDAgenziaConcorrente As New CCursorField(Of Integer)("IDAgenziaConcorrente")
        Private m_NomeAgenziaConcorrente As New CCursorFieldObj(Of String)("NomeAgenziaConcorrente")
        Private m_NomeProdottoConcorrente As New CCursorFieldObj(Of String)("NomeProdottoConcorrente")
        Private m_NumeroPreventivoConcorrente As New CCursorFieldObj(Of String)("NumeroPreventivoConcorrente")
        Private m_RataConcorrente As New CCursorField(Of Decimal)("RataConcorrente")
        Private m_DurataConcorrente As New CCursorField(Of Integer)("DurataConcorrente")
        Private m_NettoRicavoConcorrente As New CCursorField(Of Decimal)("NettoRicavoConcorrente")
        Private m_TANConcorrente As New CCursorField(Of Double)("TANConcorrente")
        Private m_TAEGConcorrente As New CCursorField(Of Double)("TAEGConcorrente")
        Private m_IDOffertaIniziale As New CCursorField(Of Integer)("IDOffertaIniziale")
        Private m_InviatoA As New CCursorFieldObj(Of String)("InviatoA")
        Private m_InviatoACC As New CCursorFieldObj(Of String)("InviatoACC")
        Private m_MezzoDiInvio As New CCursorFieldObj(Of String)("MezzoDiInvio")
        Private m_SendSubject As New CCursorFieldObj(Of String)("SendSubject")
        Private m_SendMessange As New CCursorFieldObj(Of String)("SendMessange")
        Private m_SendDate As New CCursorField(Of Date)("SendDate")
        Private m_RicevutoIl As New CCursorField(Of Date)("RicevutoIl")
        Private m_RispostoIl As New CCursorField(Of Date)("RispostoIl")
        Private m_RispostoDa As New CCursorFieldObj(Of String)("RispostoDa")
        Private m_RispostoAMezzo As New CCursorFieldObj(Of String)("RispostoAMezzo")
        Private m_RispostoSubject As New CCursorFieldObj(Of String)("RispostoSubject")
        Private m_RispostoMessage As New CCursorFieldObj(Of String)("RispostoMessage")
        Private m_IDOffertaCorrente As New CCursorField(Of Integer)("IDOffertaCorrente")
        Private m_Flags As New CCursorField(Of Integer)("Flags")
        Private m_IDFinestraLavorazione As New CCursorField(Of Integer)("IDFinestraLavorazione")

        Public Sub New()
        End Sub

        Public ReadOnly Property IDCliente As CCursorField(Of Integer)
            Get
                Return Me.m_IDCliente
            End Get
        End Property

        Public ReadOnly Property NomeCliente As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeCliente
            End Get
        End Property

        Public ReadOnly Property StatoRichiesta As CCursorField(Of StatoRichiestaDeroga)
            Get
                Return Me.m_StatoRichiesta
            End Get
        End Property


        Public ReadOnly Property DataRichiesta As CCursorField(Of Date)
            Get
                Return Me.m_DataRichiesta
            End Get
        End Property

        Public ReadOnly Property IDRichiedente As CCursorField(Of Integer)
            Get
                Return Me.m_IDRichiedente
            End Get
        End Property

        Public ReadOnly Property NomeRichiedente As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeRichiedente
            End Get
        End Property

        Public ReadOnly Property MotivoRichiesta As CCursorFieldObj(Of String)
            Get
                Return Me.m_MotivoRichiesta
            End Get
        End Property

        Public ReadOnly Property IDAgenziaConcorrente As CCursorField(Of Integer)
            Get
                Return Me.m_IDAgenziaConcorrente
            End Get
        End Property

        Public ReadOnly Property NomeAgenziaConcorrente As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeAgenziaConcorrente
            End Get
        End Property

        Public ReadOnly Property NomeProdottoConcorrente As CCursorFieldObj(Of String)
            Get
                Return Me.m_NomeProdottoConcorrente
            End Get
        End Property

        Public ReadOnly Property NumeroPreventivoConcorrente As CCursorFieldObj(Of String)
            Get
                Return Me.m_NumeroPreventivoConcorrente
            End Get
        End Property

        Public ReadOnly Property RataConcorrente As CCursorField(Of Decimal)
            Get
                Return Me.m_RataConcorrente
            End Get
        End Property

        Public ReadOnly Property DurataConcorrente As CCursorField(Of Integer)
            Get
                Return Me.m_DurataConcorrente
            End Get
        End Property

        Public ReadOnly Property NettoRicavoConcorrente As CCursorField(Of Decimal)
            Get
                Return Me.m_NettoRicavoConcorrente
            End Get
        End Property

        Public ReadOnly Property TANConcorrente As CCursorField(Of Double)
            Get
                Return Me.m_TANConcorrente
            End Get
        End Property

        Public ReadOnly Property TAEGConcorrente As CCursorField(Of Double)
            Get
                Return Me.m_TAEGConcorrente
            End Get
        End Property

        Public ReadOnly Property IDOffertaIniziale As CCursorField(Of Integer)
            Get
                Return Me.m_IDOffertaIniziale
            End Get
        End Property

        Public ReadOnly Property InviatoA As CCursorFieldObj(Of String)
            Get
                Return Me.m_InviatoA
            End Get
        End Property

        Public ReadOnly Property InviatoACC As CCursorFieldObj(Of String)
            Get
                Return Me.m_InviatoACC
            End Get
        End Property

        Public ReadOnly Property MezzoDiInvio As CCursorFieldObj(Of String)
            Get
                Return Me.m_MezzoDiInvio
            End Get
        End Property

        Public ReadOnly Property SendSubject As CCursorFieldObj(Of String)
            Get
                Return Me.m_SendSubject
            End Get
        End Property

        Public ReadOnly Property SendMessage As CCursorFieldObj(Of String)
            Get
                Return Me.m_SendMessange
            End Get
        End Property

        Public ReadOnly Property SendDate As CCursorField(Of Date)
            Get
                Return Me.m_SendDate
            End Get
        End Property

        Public ReadOnly Property RicevutoIl As CCursorField(Of Date)
            Get
                Return Me.m_RicevutoIl
            End Get
        End Property

        Public ReadOnly Property RispostoIl As CCursorField(Of Date)
            Get
                Return Me.m_RispostoIl
            End Get
        End Property

        Public ReadOnly Property RispostoDa As CCursorFieldObj(Of String)
            Get
                Return Me.m_RispostoDa
            End Get
        End Property

        Public ReadOnly Property RispostoAMezzo As CCursorFieldObj(Of String)
            Get
                Return Me.m_RispostoAMezzo
            End Get
        End Property

        Public ReadOnly Property RispostoSubject As CCursorFieldObj(Of String)
            Get
                Return Me.m_RispostoSubject
            End Get
        End Property

        Public ReadOnly Property RispostoMessage As CCursorFieldObj(Of String)
            Get
                Return Me.m_RispostoMessage
            End Get
        End Property

        Public ReadOnly Property IDOffertaCorrente As CCursorField(Of Integer)
            Get
                Return Me.m_IDOffertaCorrente
            End Get
        End Property

        Public ReadOnly Property Flags As CCursorField(Of Integer)
            Get
                Return Me.m_Flags
            End Get
        End Property

        Public ReadOnly Property IDFinestraLavorazione As CCursorField(Of Integer)
            Get
                Return Me.m_IDFinestraLavorazione
            End Get
        End Property



        Protected Overrides Function GetConnection() As CDBConnection
            Return Finanziaria.Database
        End Function

        Protected Overrides Function GetModule() As CModule
            Return Finanziaria.RichiesteDeroghe.Module
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_RichiesteDeroghe"
        End Function
    End Class


End Class

File: MiniDOMFinanziaria/Finanziaria/RichiesteDeroghe/CRichiestaDerogaCursor.cs
using System;

namespace minidom
{
    public partial class Finanziaria
    {
        public class CRichiestaDerogaCursor : Databases.DBObjectCursorPO<CRichiestaDeroga>
        {
            private DBCursorField<int> m_IDCliente = new DBCursorField<int>("IDCliente");
            private DBCursorStringField m_NomeCliente = new DBCursorStringField("NomeCliente");
            private DBCursorField<StatoRichiestaDeroga> m_StatoRichiesta = new DBCursorField<StatoRichiestaDeroga>("StatoRichiesta");
            private DBCursorField<DateTime> m_DataRichiesta = new DBCursorField<DateTime>("DataRichiesta");
            private DBCursorField<int> m_IDRichiedente = new DBCursorField<int>("IDRichiedente");
            private DBCursorStringField m_NomeRichiedente = new DBCursorStringField("NomeRichiedente");
            private DBCursorStringField m_MotivoRichiesta = new DBCursorStringField("MotivoRichiesta");
            private DBCursorField<int> m_IDAgenziaConcorrente = new DBCursorField<int>("IDAgenziaConcorrente");
            private DBCursorStringField m_NomeAgenziaConcorrente = new DBCursorStringField("NomeAgenziaConcorrente");
            private DBCursorStringField m_NomeProdottoConcorrente = new DBCursorStringField("NomeProdottoConcorrente");
            private DBCursorStringField m_NumeroPreventivoConcorrente = new DBCursorStringField("NumeroPreventivoConcorrente");
            private DBCursorField<decimal> m_RataConcorrente = new DBCursorField<decimal>("RataConcorrente");
            private DBCursorField<int> m_DurataConcorrente = new DBCursorField<int>("DurataConcorrente");
            private DBCursorField<decimal> m_NettoRicavoConcorrente = new DBCursorField<decimal>("NettoRicavoConcorrente");
            private DBCursorField<double> m_TANConcorrente = new DBCursorField<double>("TANConcorrente");
            private DBCursorField<double> m_TAEGConcorrente = new DBCursorField<double>("TAEGConcorrente");
            private DBCursorField<int> m_IDOffertaIniziale = new DBCursorField<int>("IDOffertaIniziale");
            private DBCursorStringField m_InviatoA = new DBCursorStringField("InviatoA");
            private DBCursorStringField m_InviatoACC = new DBCursorStringField("InviatoACC");
            private DBCursorStringField m_MezzoDiInvio = new DBCursorStringField("MezzoDiInvio");
            private DBCursorStringField m_SendSubject = new DBCursorStringField("SendSubject");
            private DBCursorStringField m_SendMessange = new DBCursorStringField("SendMessange");
            private DBCursorField<DateTime> m_SendDate = new DBCursorField<DateTime>("SendDate");
            private DBCursorField<DateTime> m_RicevutoIl = new DBCursorField<DateTime>("RicevutoIl");
            private DBCursorField<DateTime> m_RispostoIl = new DBCursorField<DateTime>("RispostoIl");
            private DBCursorStringField m_RispostoDa = new DBCursorStringField("RispostoDa");
            private DBCursorStringField m_RispostoAMezzo = new DBCursorStringField("RispostoAMezzo");
            private DBCursorStringField m_RispostoSubject = new DBCursorStringField("RispostoSubject");
            private DBCursorStringField m_RispostoMessage = new DBCursorStringField("RispostoMessage");
            private DBCursorField<int> m_IDOffertaCorrente = new DBCursorField<int>("IDOffertaCorrente");
            private DBCursorField<int> m_Flags = new DBCursorField<int>("Flags");
            private DBCursorField<int> m_IDFinestraLavorazione = new DBCursorField<int>("IDFinestraLavorazione");

            public CRichiestaDerogaCursor()
            {
            }

            public DBCursorField<int> IDCliente
            {
                get
                {
                    return m_IDCliente;
                }
            }

            public DBCursorStringField NomeCliente
            {
                get
                {
                    return m_NomeCliente;
                }
            }

            public DBCursorField<StatoRichiestaDeroga> StatoRichiesta
            {
                get
                {
                    return m_StatoRichiesta;
                }
            }

            public DBCursorField<DateTime> DataRichiesta
            {
                get
                {
                    return m_DataRichiesta;
                }
            }

            public DBCursorField<int> IDRichiedente
            {
                get
                {
                    return m_IDRichiedente;
                }
            }

            public DBCursorStringField NomeRichiedente
            {
                get
                {
                    return m_NomeRichiedente;
                }
            }

            public DBCursorStringField MotivoRichiesta
            {
                get
                {
                    return m_MotivoRichiesta;
                }
            }

            public DBCursorField<int> IDAgenziaConcorrente
            {
                get
                {
                    return m_IDAgenziaConcorrente;
                }
            }

            public DBCursorStringField NomeAgenziaConcorrente
            {
                get
                {
                    return m_NomeAgenziaConcorrente;
                }
            }

            public DBCursorStringField NomeProdottoConcorrente
            {
                get
                {
                    return m_NomeProdottoConcorrente;
                }
            }

            public DBCursorStringField NumeroPreventivoConcorrente
            {
                get
                {
                    return m_NumeroPreventivoConcorrente;
                }
            }

            public DBCursorField<decimal> RataConcorrente
            {
                get
                {
                    return m_RataConcorrente;
                }
            }

            public DBCursorField<int> DurataConcorrente
            {
                get
                {
                    return m_DurataConcorrente;
                }
            }

            public DBCursorField<decimal> NettoRicavoConcorrente
            {
                get
                {
                    return m_NettoRicavoConcorrente;
                }
            }

            public DBCursorField<double> TANConcorrente
            {
                get
                {
                    return m_TANConcorrente;
                }
            }

            public DBCursorField<double> TAEGConcorrente
            {
                get
                {
                    return m_TAEGConcorrente;
                }
            }

            public DBCursorField<int> IDOffertaIniziale
            {
                get
                {
                    return m_IDOffertaIniziale;
                }
            }

            public DBCursorStringField InviatoA
            {
                get
                {
                    return m_InviatoA;
                }
            }

            public DBCursorStringField InviatoACC
            {
                get
                {
                    return m_InviatoACC;
                }
            }

            public DBCursorStringField MezzoDiInvio
            {
                get
                {
                    return m_MezzoDiInvio;
                }
            }

            public DBCursorStringField SendSubject
            {
                get
                {
                    return m_SendSubject;
                }
            }

            public DBCursorStringField SendMessage
            {
                get
                {
                    return m_SendMessange;
                }
            }

            public DBCursorField<DateTime> SendDate
            {
                get
                {
                    return m_SendDate;
                }
            }

            public DBCursorField<DateTime> RicevutoIl
            {
                get
                {
                    return m_RicevutoIl;
                }
            }

            public DBCursorField<DateTime> RispostoIl
            {
                get
                {
                    return m_RispostoIl;
                }
            }

            public DBCursorStringField RispostoDa
            {
                get
                {
                    return m_RispostoDa;
                }
            }

            public DBCursorStringField RispostoAMezzo
            {
                get
                {
                    return m_RispostoAMezzo;
                }
            }

            public DBCursorStringField RispostoSubject
            {
                get
                {
                    return m_RispostoSubject;
                }
            }

            public DBCursorStringField RispostoMessage
            {
                get
                {
                    return m_RispostoMessage;
                }
            }

            public DBCursorField<int> IDOffertaCorrente
            {
                get
                {
                    return m_IDOffertaCorrente;
                }
            }

            public DBCursorField<int> Flags
            {
                get
                {
                    return m_Flags;
                }
            }

            public DBCursorField<int> IDFinestraLavorazione
            {
                get
                {
                    return m_IDFinestraLavorazione;
                }
            }

            protected override Databases.CDBConnection GetConnection()
            {
                return Database;
            }

            protected override Sistema.CModule GetModule()
            {
                return RichiesteDeroghe.Module;
            }

            public override string GetTableName()
            {
                return "tbl_RichiesteDeroghe";
            }
        }
    }
}